crates/schema/src/schema.rs

```diff
@@ -46,23 +49,41 @@ trait Schema: Sized {
 pub struct TableSchema {
     /// The unique identifier of the table within the database.
     pub table_id: TableId,
+
     /// The name of the table.
     pub table_name: Box<str>,
+
     /// The columns of the table.
     /// Inaccessible to prevent mutation.
     /// The ordering of the columns is significant. Columns are frequently identified by `ColId`, that is, position in this list.
     columns: Vec<ColumnSchema>,
+
+    /// The primary key of the table, if present. Must refer to a valid column.
+    ///
+    /// Currently, there must be a unique constraint and an index corresponding to the primary key.
+    /// Eventually, we may remove the requirement for an index.
+    ///
+    /// The database engine does not actually care about this, but client code generation does.
+    pub primary_key: Option<ColId>,
+
```

Comment:
I think this should instead be a method that returns the `pk` from the `constraints`. There is a good reason to cache this?

Reply:
There no longer are any `Constraints` stored in the TableSchema. This is the single source of truth for the primary key.